### PR TITLE
[SPARK-14639][PYTHON][R] Add `bround` function in Python/R.

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
## What changes were proposed in this pull request?

This issue aims to expose Scala `bround` function in Python/R API.
`bround` function is implemented in SPARK-14614 by extending current `round` function.
We used the following semantics from Hive.

``` java
public static double bround(double input, int scale) {
    if (Double.isNaN(input) || Double.isInfinite(input)) {
      return input;
    }
    return BigDecimal.valueOf(input).setScale(scale, RoundingMode.HALF_EVEN).doubleValue();
}
```

After this PR, `pyspark` and `sparkR` also support `bround` function.

**PySpark**

``` python
>>> from pyspark.sql.functions import bround
>>> sqlContext.createDataFrame([(2.5,)], ['a']).select(bround('a', 0).alias('r')).collect()
[Row(r=2.0)]
```

**SparkR**

``` r
> df = createDataFrame(sqlContext, data.frame(x = c(2.5, 3.5)))
> head(collect(select(df, bround(df$x, 0))))
  bround(x, 0)
1            2
2            4
```
## How was this patch tested?

Pass the Jenkins tests (including new testcases).

### CODE REVIEW

File: python/pyspark/sql/functions.py [hunk 1]
```diff
@@ -467,16 +467,29 @@ def randn(seed=None):
 @since(1.5)
 def round(col, scale=0):
     """
-    Round the value of `e` to `scale` decimal places if `scale` >= 0
+    Round the given value to `scale` decimal places using HALF_UP rounding mode if `scale` >= 0
     or at integral part when `scale` < 0.
 
-    >>> sqlContext.createDataFrame([(2.546,)], ['a']).select(round('a', 1).alias('r')).collect()
-    [Row(r=2.5)]
+    >>> sqlContext.createDataFrame([(2.5,)], ['a']).select(round('a', 0).alias('r')).collect()
```
Comment (felixcheung) [inline python/pyspark/sql/functions.py]: is this changing the existing behavior?

Comment (dongjoon-hyun) [inline python/pyspark/sql/functions.py]: Thank you, @felixcheung !
I just tried to match the example between `round` and `bround`.
`round` does not change anything.

**Comment by dongjoon-hyun** [comment]
Hi, @davies . 
Could you review this PR? After your comments, I studied and finally learned how to expose function in Python and R.

And, if you don't mind, may I include some minor PR #12329 here too? That PR is so minor and needed to be merged other PR like this.

File: R/pkg/R/functions.R [hunk 2]
```diff
@@ -1008,6 +1008,24 @@ setMethod("round",
...
+#' Returns the value of the column `e` rounded to `scale` decimal places using HALF_EVEN rounding
+#' mode if `scale` >= 0 or at integral part when `scale` < 0.
+#'
+#' @rdname bround
+#' @name bround
+#' @family math_funcs
+#' @export
+#' @examples \dontrun{bround(df$c, 0)}
+setMethod("bround",
+          signature(x = "Column", scale = "numeric"),
```
Comment (felixcheung) [inline R/pkg/R/functions.R]: should scale be default to 0 to be consistent with Python?
to do that, do
```
           signature(x = "Column"),
           function(x, scale = 0),
```

Comment (dongjoon-hyun) [inline R/pkg/R/functions.R]: Oh, thank you for sharing know-how. I didn't know that how to do!
I will fix right now.

File: R/pkg/R/functions.R [hunk 3]
```diff
@@ -1008,6 +1008,24 @@ setMethod("round",
...
+#' bround
+#'
+#' Returns the value of the column `e` rounded to `scale` decimal places using HALF_EVEN rounding
+#' mode if `scale` >= 0 or at integral part when `scale` < 0.
+#'
+#' @rdname bround
+#' @name bround
+#' @family math_funcs
+#' @export
+#' @examples \dontrun{bround(df$c, 0)}
```
Comment (shivaram) [inline R/pkg/R/functions.R]: Could you give a more detailed example of where it gets rounded down and where it gets rounded up ?

Comment (dongjoon-hyun) [inline R/pkg/R/functions.R]: Sure. Thank you for review. I'll add more example.

**Comment by davies** [comment]
LGTM

**Comment by dongjoon-hyun** [comment]
Thank you, @davies !
Or, may I add the minor PR to here? Last time, I remember that you prefer that way.

**Comment by dongjoon-hyun** [comment]
I mean #12329 . I want to close that one in order to reduce committers' review & merge cost.

**Comment by davies** [comment]
Merging this into master, thanks!

**Pull request closed without merging**